lib/temporal/workflow/dispatcher.rb

```diff
@@ -1,19 +1,38 @@
 module Temporal
   class Workflow
+    # This provides a generic event dispatcher mechanism. There are two main entry
+    # points to this class, #register_handler and #dispatch.
+    #
+    # A handler may be associated with a specific event name so when that event occurs
+    # elsewhere in the system we may dispatch the event and execute the handler. By default
+    # we do a simple name <=> handler association.
+    #
+    # Optionally, we may register a named handler that is triggered when an event _and
+    # an optional handler_name key_ are provided. When this more specific match
+    # fails, we fall back to matching just against +event_name+.
+    #
```

Comment:
I feel like a cleaner solution would be to add the `handler_name` to `event_name` and dispatch the event twice:
- "signaled:signalName"
- "signaled"

I think this would keep the dispatcher simpler for now and does allow us to achieve everything that is needed